src/Controller/Schedule/Duplicate.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Schedule;
+
+use App\Repository\Schedule\ScheduleRepositoryInterface;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+
+final class Duplicate
+{
+    private $router;
+    private $repository;
+
+    public function __construct(
+        RouterInterface $router,
+        ScheduleRepositoryInterface $repository
+    ) {
+        $this->router = $router;
+        $this->repository = $repository;
+    }
+
+    /**
+     * @Security("is_granted('ROLE_ADMIN')")
+     */
+    public function handle(Request $request): Response
+    {
+        $scheduleExists = $this->repository->find($request->attributes->get('id'));
+
+        if ($scheduleExists) {
+            $this->repository->duplicate($scheduleExists);
+        }
+
+        return new RedirectResponse($this->router->generate('schedule_index'));
```

Comment:
as your route start from a schedule id, not finding it is unexpected behavior, so it's best to throw an Exception as you can find in other edit controllers:
```
if (!$scheduleExist) {
    throw new NotFoundHttpException();
} 
```
And therefore you can do for your function
```
if (!($scheduleExist = $this->repository->find($request->attributes->get('id')))) {
    throw new NotFoundHttpException();
} 
$this->repository->duplicate($scheduleExists);
return new RedirectResponse($this->router->generate('schedule_index'));
```